spring-rabbit/src/main/java/org/springframework/amqp/rabbit/annotation/Exchange.java

```diff
@@ -55,6 +55,17 @@
 	String autoDelete() default "false";
 
 	/**
+	 * @return true if the exchange is to be declared as internal.
+	 */
+	String internal() default "false";
```

Comment:
I thought so too, but then I thought some more. Let's say the admin has set up exchange to exchange binding

`foo -> bar`

where bar is an internal exchange.

The admin only allows users to bind queues to bar. Hence, in order to avoid mismatched configuration errors, we need to support `internal` on the `@Exchange`.

Now, thinking some more, from a user's perspective, I shouldn't have to care about knowing the configuration of the exchange I am binding to (except, perhaps, its type).

I was thinking of adding another option to the `@Exchange` - `existing = "true"` which means "bind to an existing exchange - don't declare it".

But in the end, I decided it was easier to simply support `ignoreDeclarationExceptions`.